Merge Meeds-io/MIPs#29

boubaker opened the pull request.

boubaker committed:
feat: Rename Default Branding Company Title and allow to extend login page titles - MEED-1303 - Meeds-io/MIPs#29 (#487)

Prior to this change, the default Branding Company Slogan was 'Employee
Recognition'. this change will update it to use 'Web3 Hub' as made on
deed-tenant addon. In addition, this change will add two labels coming
from 'exo.properties' file that allows to customize displayed labels in
login page.

<!-- Ensure to provide github issue and task id in the title -->
<!-- Choose between feat and fix in the title to differenciate a new
feature from a fix -->
<!-- Title format must be :
feat: FEATURE TITLE - MEED-XXXX - meeds-io/meeds#1234
or
fix: Fix TITLE - MEED-XXXX - meeds-io/meeds#1234
-->

<!-- Description : describe the feature/the fix by answering theses
questions : -->
<!-- Why is this change needed?-->
<!-- Prior to this change, ...-->
<!-- How does it address the issue?-->
<!-- This change ...-->


<!-- Tips : 
Try To Limit Each Line to a Maximum Of 72 Characters
Provide links or keys to any relevant tickets, articles or other
resources

Remember to
- Capitalize the subject line
- Use the imperative mood in the subject line
- Do not end the subject line with a period
- Separate subject from body with a blank line
- Use the body to explain what and why vs. how
- Can use multiple lines with "-" for bullet points in body
-->

boubaker committed:
feat: Rework Forgot Password Process to be able to use Vue Application in Front-end - MEED-1304 - Meeds-io/MIPs#29 (#491)

Prior to this change, the forgot password handler was forwarding request
to JSP standalone file without including appropriate JS & CSS Modules to
be able to display this UI with Vue & Vuetify in FrontEnd. This change
will use as parent class 'JspBasedWebHandler' to be able to include the
mandatory JS & CSS modules for a Vue based UI. In addition, some Unit
Tests has been added to make the Forgot Password Process verified.

boubaker committed:
feat: Rework Internal Onboarding Process to be able to use Vue Application in Frontend - MEED-851 - Meeds-io/MIPs#29 (#492)

Prior to this change, the internal onboarding handler was forwarding
request to JSP standalone file without including appropriate JS & CSS
Modules to be able to display this UI with Vue & Vuetify in FrontEnd.
This change will use as parent class 'JspBasedWebHandler' to be able to
include the mandatory JS & CSS modules for a Vue based UI. In addition,
some Unit Tests has been added to make the Onboarding Process more
stable.

boubaker committed:
feat: Rework External User Onboarding Registration Process to be able  to use Vue Application in Frontend - MEED-1392 - Meeds-io/MIPs#29 (#494)

Prior to this change, the external users onboarding with registration
handler was forwarding request to JSP standalone file without including
appropriate JS & CSS Modules to be able to display this UI with Vue &
Vuetify in FrontEnd. This change will use as parent class
'JspBasedWebHandler' to be able to include the mandatory JS & CSS
modules for a Vue based UI. In addition, some Unit Tests has been added
to make the Onboarding Process more stable.

boubaker committed:
feat: Rework Register Form to be able  to use Vue Application in Frontend - MEED-880 - Meeds-io/MIPs#29 (#495)

boubaker committed:
fix: Fix OAuth Missing Dependency in packaging marked as provided - MEED-1308 - Meeds-io/MIPs#29

rdenarie commented on component/web/api/src/main/java/org/exoplatform/web/filter/RedefinableResourceFilter.java:
```diff
@@ -51,6 +54,15 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             path = path.substring(ctx.length());
         }
 
+        if (StringUtils.contains(path, "favicon")) {
```
```suggestion
        if (StringUtils.contains(path, "favicon.ico")) {
```
Can we be more specific ? I'm afraid that checking on favicon string can lead to modify for resources with favicon in name, but not used as favicon

boubaker replied:
Done, thanks.

rdenarie commented on component/web/security/src/main/java/org/exoplatform/web/login/LoginHandler.java:
```diff
@@ -392,8 +392,11 @@ private void extendUIParameters(ControllerContext controllerContext, LoginStatus
 
-      if (status != LoginStatus.AUTHENTICATED && status != LoginStatus.UNAUTHENTICATED) {
-        params.put("errorCode", status.getErrorCode());
+      params.put("authenticationTitle", PropertyManager.getProperty("portal.authentication.title"));
+      params.put("authenticationSubtitle", PropertyManager.getProperty("portal.authentication.subtitle"));
```
Should theses lines be added in other handler ? 
Currently the login page does correctly use properties set in exo.properties, but page forgot password and onboarding user display Title/SubTitle

boubaker replied:
Done, thanks.

boubaker committed:
fix: Fix Extending Login Title & Subtitle to forgot password and registration pages - Meeds-io/MIPs#29
Prior to this change, the title and subtitle extension via properties wasn't applied on other pages but login page.
